### PR TITLE
k8s.io/apiserver/storage/etcd: refactor etcd GetList.

### PR DESCRIPTION
**Pull request by siyuanfoundation**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
refactor etcd GetList.
to make the GetList easier to understand.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by siyuanfoundation** [comment]
Followup of https://github.com/kubernetes/kubernetes/pull/121049

**Comment by seans3** [comment]
This is a very significant change. Please attend the bi-weekly sig api-machinery meeting to get direction for this effort first.

**Comment by siyuanfoundation** [comment]
cc @wojtek-t

File: staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go [hunk 1]
```diff
@@ -639,6 +639,108 @@ func TestInvalidKeys(t *testing.T) {
 	expectInvalidKey("Count", countErr)
 }
 
+func TestResolveGetListRev(t *testing.T) {
```
Comment (heqg) [inline staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go:642]: /lgtm

**Review by heqg — APPROVED** [review]

File: staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go [hunk 2]
```diff
@@ -639,6 +639,108 @@ func TestInvalidKeys(t *testing.T) {
...
+			expectedError: "invalid resource version",
+		},
+		{
+			name:          "unknown ResourceVersionMatch value",
+			rv:            "200",
+			rvMatch:       "unknown",
+			expectedError: "unknown ResourceVersionMatch value",
+		},
+		{
+			name:        "use continueRV",
```
Comment (wojtek-t) [inline staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go:674]: Add another testcase which is identical to this one but has RV=""

File: staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go [hunk 3]
```diff
@@ -639,6 +639,108 @@ func TestInvalidKeys(t *testing.T) {
...
+			continueKey: "continue",
+			continueRV:  100,
+			rv:          "0",
+			expectedRev: 100,
+		},
```
Comment (wojtek-t) [inline staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go:679]: And yet another one that has continueRV=0.

**Review by wojtek-t — COMMENTED** [review]
Just two minor comments about the test - other than that LGTM.

File: staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go (hunk 2 above)
Comment (siyuanfoundation) [reply staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go:674]: done

File: staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go (hunk 3 above)
Comment (siyuanfoundation) [reply staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go:679]: done

**Comment by wojtek-t** [comment]
/lgtm
/approve

Thanks!

**Pull request merged**